cmd/vaultworker/main.go

```diff
@@ -186,69 +187,75 @@ func main() {
 	if err != nil {
 		lc.Error("Fatal Error importing Admin policy in Vault.")
-		fatalIfErr(err, "Import policy failure")
+		worker.FatalIfErr(err, "Import policy failure")
 	}
 
 	// Create Admin token associated with admin policy in Vault
 	lc.Info("Creating Vault Admin token.")
-	err = createToken(config.SecretService.TokenName4Admin, config.SecretService.PolicyName4Admin, rootToken.Token, config, client)
+	err = worker.CreateToken(config.SecretService.TokenName4Admin, config.SecretService.PolicyName4Admin, rootToken.Token, config, client)
 	if err != nil {
 		lc.Error("Fatal Error creating Admin token in Vault.")
-		fatalIfErr(err, "Create token failure (Admin)")
+		worker.FatalIfErr(err, "Create token failure (Admin)")
 	}
 
 	// ------------------ Kong Vault Policies and associated token ----------------------
 	// Read the Kong HCL file and build the policy request
 	lc.Info("Verifying Kong policy file hash (SHA256).")
 	policyFile = config.SecretService.PolicyPath4Kong
-	_, err = hashFile(&policyFile, debug)
+	_, err = worker.HashFile(&policyFile, debug)
 	if err != nil {
-		fatalIfErr(err, "Calculating policy file hash (SHA256)")
+		worker.FatalIfErr(err, "Calculating policy file hash (SHA256)")
 	}
 	lc.Info("Reading Kong policy file.")
 	policyFile = config.SecretService.PolicyPath4Kong
-	policyRequest, err = getPolicyFromFile(&policyFile)
+	policyRequest, err = worker.GetPolicyFromFile(&policyFile)
 	if err != nil {
 		lc.Error("Fatal Error opening Kong policy file.")
-		fatalIfErr(err, "Opening policy file (Kong)")
+		worker.FatalIfErr(err, "Opening policy file (Kong)")
 	}
 
 	// Import the Kong policy data into Vault
 	lc.Info("Importing Vault Kong policy.")
-	err = importPolicy(config.SecretService.PolicyName4Kong, &policyRequest, rootToken.Token, config, client)
+	err = worker.ImportPolicy(config.SecretService.PolicyName4Kong, &policyRequest, rootToken.Token, config, client)
 	if err != nil {
 		lc.Error("Fatal Error importing Kong policy in Vault.")
-		fatalIfErr(err, "Import policy failure")
+		worker.FatalIfErr(err, "Import policy failure")
 	}
 
 	// Create Kong token associated with kong policy in Vault
 	lc.Info("Creating Vault Kong token.")
-	err = createToken(config.SecretService.TokenName4Kong, config.SecretService.PolicyName4Kong, rootToken.Token, config, client)
+	err = worker.CreateToken(config.SecretService.TokenName4Kong, config.SecretService.PolicyName4Kong, rootToken.Token, config, client)
 	if err != nil {
 		lc.Error("Fatal Error creating Kong token in Vault.")
-		fatalIfErr(err, "Create token failure (Kong)")
+		worker.FatalIfErr(err, "Create token failure (Kong)")
 	}
-	// -----------------------------------------------------------------------------------
 
 	secretServiceBaseURL := fmt.Sprintf("https://%s:%s/", config.SecretService.Server, config.SecretService.Port)
 
-	hasCertKeyPair, err := certKeyPairInStore(config, secretServiceBaseURL, client, debug)
+	//TODO: need to implment credential creation
```

Comment:
this is okay, normally I would ask that commented out code be removed, but I spoke with @tingyuz about this but leaving this out now is the best option